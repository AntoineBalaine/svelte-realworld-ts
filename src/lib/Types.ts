export enum SignIn_Up_Alternatives {
	SignUp = "sign-up",
	SignIn = "sign-in"
}

export enum TextEditorActions {
	create = "createarticle",
	update = "updatearticle"
}
