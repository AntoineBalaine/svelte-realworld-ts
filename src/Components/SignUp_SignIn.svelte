<script lang="ts">
	import { enhance } from "$app/forms";
	import type ApiTypes from "$lib/ApiTypes";
	import { SignIn_Up_Alternatives } from "$lib/Types";

	interface SignDta {
		path: string;
		question: string;
		buttonText: string;
	}

	const SignInDta: SignDta = {
		path: "/sign-up",
		question: "Donâ€™t have an account yet?",
		buttonText: "Sign in"
	};

	const SignUp: SignDta = {
		path: "/sign-in",
		question: "Already have an account?",
		buttonText: "Sign up"
	};

	export let SignInUp_prop: SignIn_Up_Alternatives;
	export let form:
		| ApiTypes.operations["CreateUser"]["responses"]["422"]["content"]["application/json"]
		| null = null;
	$: SignIn_or_Up = SignInUp_prop === SignIn_Up_Alternatives.SignUp ? SignUp : SignInDta;
</script>

<div class="auth-page">
	<div class="container page">
		<div class="row">
			<div class="col-md-6 offset-md-3 col-xs-12">
				<h1 class="text-xs-center" data-testid="signInUp-title">{SignIn_or_Up.buttonText}</h1>
				<p class="text-xs-center">
					<a href={SignIn_or_Up.path} data-testid="signInUp-switch">{SignIn_or_Up.question}</a>
				</p>
				{#if form}
					<ul class="error-messages">
						{#each Object.entries(form.errors) as [key, value]}
							<li>{key} {value}</li>
						{/each}
					</ul>
				{/if}

				<form method="POST" use:enhance>
					{#if SignInUp_prop === SignIn_Up_Alternatives.SignUp}
						<fieldset class="form-group">
							<input
								data-testid="signUp-name"
								class="form-control form-control-lg"
								type="text"
								name="username"
								placeholder="Your Name"
							/>
						</fieldset>
					{/if}
					<fieldset class="form-group">
						<input
							data-testid="signUp-email"
							class="form-control form-control-lg"
							name="email"
							type="email"
							placeholder="Email"
						/>
					</fieldset>
					<fieldset class="form-group">
						<input
							data-testid="signUp-password"
							class="form-control form-control-lg"
							name="password"
							type="password"
							placeholder="Password"
						/>
					</fieldset>
					<button class="btn btn-lg btn-primary pull-xs-right">{SignIn_or_Up.buttonText}</button>
				</form>
			</div>
		</div>
	</div>
</div>
