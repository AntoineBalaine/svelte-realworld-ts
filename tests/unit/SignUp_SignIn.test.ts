import { render } from "@testing-library/svelte";
import { describe, test, expect } from "vitest";
import SignUp_SignIn from "src/components/SignUp_SignIn.svelte";
import { SignIn_Up_Alternatives } from "$lib/Types";
import type { ComponentProps } from "svelte";

describe("SignUp_SignIn Component", () => {
	test("Sign In: should render the component", () => {
		const props: ComponentProps<SignUp_SignIn> = {
			SignInUp_prop: SignIn_Up_Alternatives.SignIn
		};
		const signUpPreview = render(SignUp_SignIn, props);

		expect(signUpPreview.getByTestId("signInUp-title").innerHTML).toEqual("Sign in");
		expect(signUpPreview.getByTestId("signInUp-switch").innerHTML).toEqual(
			"Donâ€™t have an account yet?"
		);
		const name_input = (() => {
			try {
				return signUpPreview.getByPlaceholderText("Your Name");
			} catch {
				return undefined;
			}
		})();
		expect(name_input).toBeUndefined;
		expect(signUpPreview.getByPlaceholderText("Email")).not.toBeUndefined;
		expect(signUpPreview.getByPlaceholderText("Password")).not.toBeUndefined;

		expect(signUpPreview.container).toMatchSnapshot();
	});
	test("Sign Up: should render the component", () => {
		const props: ComponentProps<SignUp_SignIn> = {
			SignInUp_prop: SignIn_Up_Alternatives.SignUp
		};
		const signUpPreview = render(SignUp_SignIn, props);

		expect(signUpPreview.getByTestId("signInUp-title").innerHTML).toEqual("Sign up");
		expect(signUpPreview.getByTestId("signInUp-switch").innerHTML).toEqual(
			"Already have an account?"
		);
		expect(signUpPreview.getByPlaceholderText("Your Name")).not.toBeUndefined;
		expect(signUpPreview.getByPlaceholderText("Email")).not.toBeUndefined;
		expect(signUpPreview.getByPlaceholderText("Password")).not.toBeUndefined;

		expect(signUpPreview.container).toMatchSnapshot();
	});
});
