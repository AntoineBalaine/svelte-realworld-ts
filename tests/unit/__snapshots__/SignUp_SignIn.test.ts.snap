// Vitest Snapshot v1

exports[`SignUp_SignIn Component > Sign In: should render the component 1`] = `
<body>
  <div>
    <div
      class="auth-page"
    >
      <div
        class="container page"
      >
        <div
          class="row"
        >
          <div
            class="col-md-6 offset-md-3 col-xs-12"
          >
            <h1
              class="text-xs-center"
              data-testid="signInUp-title"
            >
              Sign in
            </h1>
             
            <p
              class="text-xs-center"
            >
              <a
                data-testid="signInUp-switch"
                href="/sign-up"
              >
                Donâ€™t have an account yet?
              </a>
            </p>
             
             
            <form
              method="POST"
            >
               
              <fieldset
                class="form-group"
              >
                <input
                  class="form-control form-control-lg"
                  data-testid="signUp-email"
                  name="email"
                  placeholder="Email"
                  type="email"
                />
              </fieldset>
               
              <fieldset
                class="form-group"
              >
                <input
                  class="form-control form-control-lg"
                  data-testid="signUp-password"
                  name="password"
                  placeholder="Password"
                  type="password"
                />
              </fieldset>
               
              <button
                class="btn btn-lg btn-primary pull-xs-right"
              >
                Sign in
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
    <!--&lt;SignUp_SignIn&gt;-->
  </div>
</body>
`;

exports[`SignUp_SignIn Component > Sign Up: should render the component 1`] = `
<body>
  <div>
    <div
      class="auth-page"
    >
      <div
        class="container page"
      >
        <div
          class="row"
        >
          <div
            class="col-md-6 offset-md-3 col-xs-12"
          >
            <h1
              class="text-xs-center"
              data-testid="signInUp-title"
            >
              Sign up
            </h1>
             
            <p
              class="text-xs-center"
            >
              <a
                data-testid="signInUp-switch"
                href="/sign-in"
              >
                Already have an account?
              </a>
            </p>
             
             
            <form
              method="POST"
            >
              <fieldset
                class="form-group"
              >
                <input
                  class="form-control form-control-lg"
                  data-testid="signUp-name"
                  name="username"
                  placeholder="Your Name"
                  type="text"
                />
              </fieldset>
               
              <fieldset
                class="form-group"
              >
                <input
                  class="form-control form-control-lg"
                  data-testid="signUp-email"
                  name="email"
                  placeholder="Email"
                  type="email"
                />
              </fieldset>
               
              <fieldset
                class="form-group"
              >
                <input
                  class="form-control form-control-lg"
                  data-testid="signUp-password"
                  name="password"
                  placeholder="Password"
                  type="password"
                />
              </fieldset>
               
              <button
                class="btn btn-lg btn-primary pull-xs-right"
              >
                Sign up
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
    <!--&lt;SignUp_SignIn&gt;-->
  </div>
</body>
`;
